test/framework/qa/rest-test-test/src/yamlRestTest/java/org/elasticsearch/rest/test/ClientYamlTestSuiteIT.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.rest.test;
+
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+import org.elasticsearch.test.rest.yaml.ClientYamlTestCandidate;
+import org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase;
+
+public class ClientYamlTestSuiteIT extends ESClientYamlSuiteTestCase {
```

Comment:
We have a number of tests in the `:test:framework` project itself to exercising the test framework. Is there a reason we can't put this stuff there? It seems like a much more appropriate place than introducing a new project to test a very specific piece of functionality that lives somewhere else. It would seem to me we could write an internal cluster test for this, which would remove the need to for an explicit plugin project just to declare a mock endpoint.

Reply:
Looking more at what currently exists in terms of test coverage, I think just adding unit tests to the exist suite for this is sufficient. I think end-to-end testing with an external cluster is overkill here given we have no such equivalent testing for any of the other yaml rest testing functionality.

Reply:
> I think end-to-end testing with an external cluster is overkill 

Fair point especially since there little development in this area. Will remove the additional qa project. The unit tests are already included and provide pretty comprehensive coverage.

Reply:
done on [`71f7b99` (#69501)](https://github.com/elastic/elasticsearch/pull/69501/commits/71f7b9936f0b62c59b14b7d926162dc15ae6d1b7)